src/cds/src/main/kotlin/org/icpclive/cds/cats/CATSDataSource.kt

```diff
@@ -0,0 +1,222 @@
+package org.icpclive.cds.cats
+
+import kotlinx.coroutines.*
+import kotlinx.datetime.Clock
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toKotlinLocalDateTime
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.icpclive.api.*
+import org.icpclive.cds.ContestParseResult
+import org.icpclive.cds.FullReloadContestDataSource
+import org.icpclive.cds.common.jsonLoaderService
+import org.icpclive.util.getLogger
+import java.awt.Color
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.seconds
+
+
+class CATSDataSource(val properties: Properties, creds: Map<String, String>) : FullReloadContestDataSource(5.seconds) {
+    private val login = creds["cats.login"]
```

Comment:
It's better to be consistent with other cds's. There is some wrapper function `properties.getCredentials("login", creds). 

After it you can use `login=$creds.cats_login` and `password=$creds.cats_password` in properties file.